src/main/java/com/iota/iri/service/garbagecollector/GarbageCollector.java

```diff
@@ -0,0 +1,323 @@
+        for(Map.Entry<Class<? extends GarbageCollectorJob>, ArrayDeque<GarbageCollectorJob>> entry : garbageCollectorJobs.entrySet()) {
+            if(shuttingDown) {
+                return;
+            }
+
+            QueueProcessor queueProcessor = queueProcessors.get(entry.getKey());
+            if(queueProcessor == null) {
+                throw new GarbageCollectorException("could not determine a queue processor for cleanup job of type " + entry.getKey().getCanonicalName());
+            }
+
+            queueProcessor.processQueue(this, entry.getValue());
+        }
+    }
+
+    /**
+     * This method retrieves the job queue belonging to a given job type.
+     *
+     * It first checks if a corresponding queue exists already and creates a new one if no queue was created yet for the
+     * given job type.
+     *
+     * @param jobClass type of the job that we want to retrieve the queue for
+     * @return the list of jobs for the provided job type
+     */
+    private ArrayDeque<GarbageCollectorJob> getJobQueue(Class<? extends GarbageCollectorJob> jobClass) {
+        if (garbageCollectorJobs.get(jobClass) == null) {
+            synchronized(this) {
+                if (garbageCollectorJobs.get(jobClass) == null) {
+                    garbageCollectorJobs.put(jobClass, new ArrayDeque<>());
+                }
+            }
+        }
+
+        return garbageCollectorJobs.get(jobClass);
+    }
+
+    /**
+     * This method tries to restore the previous state of the Garbage Collector by reading the state file that get's
+     * persisted whenever we modify any of the queues.
+     *
+     * It is used to restore the state of the garbage collector between IRI restarts and speed up the pruning
+     * operations. If it fails to restore the state it just continues with an empty state which doesn't cause any
+     * problems with future jobs other than requiring them to perform unnecessary steps and therefore slowing them down
+     * a bit.
+     */
+    private void restoreCleanupJobs() {
+        try {
+            BufferedReader reader = new BufferedReader(
+                new InputStreamReader(
+                    new BufferedInputStream(
+                        new FileInputStream(getStateFile())
+                    )
+                )
+            );
+
+            String line;
+            while((line = reader.readLine()) != null) {
+                String[] parts = line.split(";", 2);
+                if(parts.length >= 2) {
+                    JobParser jobParser = this.jobParsers.get(parts[0]);
+                    if(jobParser == null) {
+                        throw new GarbageCollectorException("could not determine a parser for cleanup job of type " + parts[0]);
+                    }
+
+                    addJob(jobParser.parse(parts[1]));
+                }
+            }
+
+            reader.close();
+        }
+        catch(IOException e) { /* do nothing */ }
```

Comment:
really?

Reply:
Yes really :) Having local snapshot files is optional - especially the first time you launch IRI after we enable this feature there will simply be no local snapshot files and you will start with the builtin snapshot instead.

If the files can not be parsed we handle the error but if there are no Local Snapshot files available it can be ignored because they were simply not created yet.